packages/nodes-from-anchor/src/v01/InstructionAccountNode.ts

```diff
@@ -1,25 +1,121 @@
-            : [instructionAccountNodeFromAnchorV01(account)],
+            ? instructionAccountNodesFromAnchorV01(allAccounts, instructionArguments, account.accounts)
+            : [instructionAccountNodeFromAnchorV01(allAccounts, instructionArguments, account)],
     );
 }
 
-export function instructionAccountNodeFromAnchorV01(idl: IdlV01InstructionAccount): InstructionAccountNode {
+export function instructionAccountNodeFromAnchorV01(
+    allAccounts: AccountNode[],
+    instructionArguments: InstructionArgumentNode[],
+    idl: IdlV01InstructionAccount,
+): InstructionAccountNode {
     const isOptional = idl.optional ?? false;
     const docs = idl.docs ?? [];
     const isSigner = idl.signer ?? false;
     const isWritable = idl.writable ?? false;
     const name = idl.name ?? '';
-    let defaultValue = undefined;
+    let defaultValue: PdaValueNode | PublicKeyValueNode | undefined;
 
     if (idl.address) {
         defaultValue = publicKeyValueNode(idl.address, name);
+    } else if (idl.pda) {
+        const [seeds, lookups] = idl.pda.seeds.reduce(
+            ([seeds, lookups], seed: IdlV01Seed) => {
+                const kind = seed.kind;
+
+                switch (kind) {
+                    case 'const':
+                        return [[...seeds, constantPdaSeedNodeFromBytes('base16', hex(seed.value))], lookups];
+                    case 'account': {
+                        const path = seed.path.split('.');
+                        if (path.length === 1) {
+                            return [
+                                [...seeds, variablePdaSeedNode(seed.path, publicKeyTypeNode())],
+                                [...lookups, pdaSeedValueNode(seed.path, accountValueNode(seed.path))],
+                            ];
+                        } else if (path.length === 2) {
+                            const accountName = camelCase(seed.account ?? '');
+                            const accountNode = allAccounts.find(({ name }) => name === accountName);
+                            if (!accountNode) {
+                                throw new KinobiError(KINOBI_ERROR__ANCHOR__ACCOUNT_TYPE_MISSING, { kind });
+                            }
+
+                            const fieldName = camelCase(path[1]);
+                            const accountFields = resolveNestedTypeNode(accountNode.data).fields;
+                            const fieldNode = accountFields.find(({ name }) => name === fieldName);
+                            if (!fieldNode) {
+                                throw new KinobiError(KINOBI_ERROR__ANCHOR__TYPE_PATH_MISSING, {
+                                    idlType: seed.account,
+                                    path: seed.path,
+                                });
+                            }
+
+                            const seedName = camelCase(seed.path);
+                            return [[...seeds, variablePdaSeedNode(seedName, fieldNode.type)], []];
```

Comment:
@kespinola I am a bit concerned about this use-case because we are identifying a variable seed for the PDA but we're not able to provide a value for it since the nested account isn't in the instruction accounts.

I initially tried to use `path[0]` as an `accountValueNode` here but then realised that this is not the account we should use, this is just the account that should contain the seed inside of its data which we do not have access to.

So in practice, generating a client for this use-case will throw an error because we'll create a PDA without all the required seeds. I'm not too sure if to leave this scenario as-is and revisit in the future or if to stop supporting this scenario altogether to avoid people getting confused. Wdyt?

Reply:
In this case my thought was to map thise to a variable of the instruction to be provided. It is a very common pattern in anchor IDLs so even if doesn't have auto-lookup it should be skipped gracefully.

I am also fine with dropping any generation of seeds since they aren't mapped to a top level account definition like in the shank IDL spec.

Reply:
Ah I see yeah we could use an `extraArgument` for this purpose. For now though how would we gracefully skip this scenario? Should we just discard the whole seed default value if we encounter a seed with a path length greater than 1?

Reply:
Yes. that seems appropriate.

Reply:
I've updated the PR accordingly. Let me know if that works for you.